clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -148,6 +150,33 @@ public boolean matches(ClientRequest request) {
     }
 
     @Test
+    public void testFetchedRecordsRaisesOnSerializationErrors() {
+        ByteArrayDeserializer deserializer = new ByteArrayDeserializer() {
+            @Override
+            public byte[] deserialize(String topic, byte[] data) {
+                throw new SerializationException();
```

Comment:
Interesting point. I feel that attempting to return partial fetch data may be overkill and would be awkward to implement, but I agree that surfacing the offset that failed to deserialize would be helpful. One option I can think of would be to create a `ConsumerSerializationException` which extends `SerializationException` and includes the topic, partition, and offset as accessible fields. This should work with any users attempting to catch `SerializationException`.